internal/services/api/reset_credentials_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2024 by Posit Software, PBC.
+
+package api
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/posit-dev/publisher/internal/credentials"
+	"github.com/posit-dev/publisher/internal/credentials/credentialstest"
+	"github.com/posit-dev/publisher/internal/logging"
+	"github.com/posit-dev/publisher/internal/util/utiltest"
+	"github.com/stretchr/testify/suite"
+)
+
+type ResetCredsSuite struct {
+	utiltest.Suite
+	log          logging.Logger
+	credservice  *credentialstest.CredentialsServiceMock
+	credsFactory credentials.CredServiceFactory
+}
+
+func TestResetCredsSuite(t *testing.T) {
+	suite.Run(t, new(ResetCredsSuite))
+}
+
+func (s *ResetCredsSuite) SetupTest() {
+	s.log = logging.New()
+	s.credservice = credentialstest.NewCredentialsServiceMock()
+	s.credsFactory = func(log logging.Logger) (credentials.CredentialsService, error) {
+		return s.credservice, nil
+	}
+}
+
+func (s *ResetCredsSuite) TestResetOk() {
+	path := "http://example.com/api/credentials"
+	req, err := http.NewRequest("DELETE", path, nil)
+	s.NoError(err)
+
+	s.credservice.On("Reset").Return("backup-file", nil)
+
+	rec := httptest.NewRecorder()
+	h := ResetCredentialsHandlerFunc(s.log, s.credsFactory)
+	h(rec, req)
+
+	s.Equal(http.StatusOK, rec.Result().StatusCode)
+	s.Contains(rec.Body.String(), `{"backupFile":"backup-file"}`)
+}
+
+func (s *ResetCredsSuite) TestReset_EvenWithLoadError() {
+	path := "http://example.com/api/credentials"
+	req, err := http.NewRequest("DELETE", path, nil)
+	s.NoError(err)
+
+	// Even if factory returns a LoadError,
+	// reset proceeds, that is the reason we are resetting data.
+	s.credsFactory = func(log logging.Logger) (credentials.CredentialsService, error) {
+		return s.credservice, credentials.NewLoadError(errors.New("load errsss"))
+	}
+
+	s.credservice.On("Reset").Return("", nil)
+
+	rec := httptest.NewRecorder()
+	h := ResetCredentialsHandlerFunc(s.log, s.credsFactory)
+	h(rec, req)
+
+	s.Equal(http.StatusOK, rec.Result().StatusCode)
+}
+
+func (s *ResetCredsSuite) TestReset_EvenWithCorruptedError() {
+	path := "http://example.com/api/credentials"
+	req, err := http.NewRequest("DELETE", path, nil)
+	s.NoError(err)
+
+	// Even if factory returns a CorruptedError,
+	// reset proceeds, that is the reason we are resetting data.
+	s.credsFactory = func(log logging.Logger) (credentials.CredentialsService, error) {
+		return s.credservice, credentials.NewCorruptedError("qwerty")
+	}
+
+	s.credservice.On("Reset").Return("", nil)
+
+	rec := httptest.NewRecorder()
+	h := ResetCredentialsHandlerFunc(s.log, s.credsFactory)
+	h(rec, req)
+
+	s.Equal(http.StatusOK, rec.Result().StatusCode)
+}
+
+func (s *ResetCredsSuite) TestReset_UnknownError() {
```

Comment:
If my suggestion to handle the raw file errors from the reset operations is implemented, then we should be able to make this a known error always, with the unknown being subjugated to a subError within the type.